tests/test_base_plugin.py

```diff
@@ -0,0 +1,282 @@
+        with self.assertRaises(PanoptesPluginConfigurationError):
+            panoptes_plugin_info.configMtime
+
+    def test_plugin_info_last_properties(self):
+        panoptes_plugin_info = PanoptesPluginInfo("Test Polling Plugin",
+                                                  "tests/plugins/polling/test")
+
+        #  Test last_results and last_executed return 0 on exception.
+        with self.assertRaises(AssertionError):
+            panoptes_plugin_info.last_executed = "test"
+        self.assertEqual(panoptes_plugin_info.last_executed, 0)
+
+        with self.assertRaises(AssertionError):
+            panoptes_plugin_info.last_results = "test"
+        self.assertEqual(panoptes_plugin_info.last_results, 0)
+
+    def test_plugin_info_properties(self):
+        panoptes_plugin_info = PanoptesPluginInfo("Test Polling Plugin",
+                                                  "tests/plugins/polling/test")
+
+        with self.assertRaises(PanoptesPluginConfigurationError):
+            panoptes_plugin_info.panoptes_context
+
+        panoptes_plugin_info.panoptes_context = self.__panoptes_context
+        panoptes_plugin_info.kv_store_class = PanoptesTestKeyValueStore
+        panoptes_plugin_info.last_executed = _LAST_EXECUTED_TEST_VALUE
+        panoptes_plugin_info.config_filename = "tests/plugins/polling/test/plugin_polling_test.panoptes-plugin.example"
+
+        #  Test results_cache_age, execute_frequency, last_executed return 0 on exception.
+        self.assertEqual(panoptes_plugin_info.results_cache_age, 0)
+        self.assertEqual(panoptes_plugin_info.execute_frequency, 0)
+
+        panoptes_plugin_info.details.read(panoptes_plugin_info.config_filename)
+
+        self.assertEqual(panoptes_plugin_info.results_cache_age, 100)
+        self.assertEqual(panoptes_plugin_info.execute_frequency, 60)
+
+        #  Test last_executed setter cannot be reset after being read
+        self.assertEqual(panoptes_plugin_info.last_executed, _LAST_EXECUTED_TEST_VALUE)
+        panoptes_plugin_info.last_executed = 1
+        self.assertNotEqual(panoptes_plugin_info.last_executed, 1)
+        self.assertEqual(panoptes_plugin_info.last_executed, _LAST_EXECUTED_TEST_VALUE)
+
+        with patch("yahoo_panoptes.framework.plugins.panoptes_base_plugin.time.time", mock_time):
+            self.assertEqual(panoptes_plugin_info.last_executed_age, int(_TIMESTAMP - _LAST_EXECUTED_TEST_VALUE))
+
+    def test_plugin_info_last_results(self):
+        panoptes_plugin_info = PanoptesPluginInfo("plugin_name", "path/to/plugin")
+        panoptes_plugin_info.panoptes_context = self.__panoptes_context
+        panoptes_plugin_info.kv_store_class = PanoptesTestKeyValueStore
+
+        panoptes_plugin_info.last_results = _LAST_RESULTS_TEST_VALUE
+        self.assertEqual(panoptes_plugin_info.last_results, _LAST_RESULTS_TEST_VALUE)
+
+        # Test value is cached.
+        self.assertEqual(panoptes_plugin_info.last_results, _LAST_RESULTS_TEST_VALUE)
```

Comment:
We probably need to mock and fail the Redis connection and ensure that this continues to work.

Reply:
Moved lines 189, 190 (and 205, 206) inside the context manager to make this test clearer.